pkg/ipamd/datastore/data_store.go

```diff
@@ -1238,3 +1238,36 @@ func (ds *DataStore) FindFreeableCidrs(eniID string) []CidrInfo {
 func DivCeil(x, y int) int {
 	return (x + y - 1) / y
 }
+
+// CheckFreeableENIexists will return true if there is an ENI which is unused.
+// Could have just called getDeletbleENI, this is just to optimize a bit.
+func (ds *DataStore) CheckFreeableENIexists() bool {
```

Comment:
Ok, don't we already have an equivalent function? (i.e.,) getDeletableENI? I see the only additional checks in that function are for WARM IP/Prefix targets and we need them anyways right? Can you expand a bit on the need for this?

Reply:
Yes I agree but even before calling `tryFreeENI` which eventually calls `getDeletableENI` , I am just checking if the ENI has some pods. This is just a minor optimization because we call this function every reconcile time and `getDeletableENI` does lot more work to check if it is needed for warm targets.

Reply:
Right but we can't anyways free the ENI without those checks. Are you saying this will be called much more frequently? Maybe we can replace the initial part of `getDeletableENI` with this function so that we can add any additional conditions in this function alone.

typo in comment: getDeletbleENI -> getDeletableENI

Reply:
Right this will be called much more frequently because of the reconciler. Yes that be done.